File: Kth_Largest_Element_in_an_Array_v1.java
import java.util.Arrays;
import java.util.Collections;

class Solution {
    public int findKthLargest(int[] nums, int k) {
        Arrays.sort(nums);
        return nums[nums.length - k];
    }
}

public class Kth_Largest_Element_in_an_Array_v1 {
    public static void main(String[] args) {
        Solution solution = new Solution();
        int[][] testset = { { 3, 2, 1, 5, 6, 4 }, { 3, 2, 3, 1, 2, 4, 5, 5, 6 } };
        int[] ks = { 2, 4 };
        for (int i = 0; i < testset.length; i++) {
            System.out.println(solution.findKthLargest(testset[i], ks[i]));
        }
    }
}

File: Kth_Largest_Element_in_an_Array_v1.cpp
#include <iostream>
#include <queue>
#include <vector>
using namespace std;

// 用小顶堆维护前 k 个最大元素，堆顶即为第 k 大元素
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        priority_queue<int, vector<int>, greater<int>> pq;
        for (int i : nums) {
            if (pq.size() < k) {
                pq.push(i);
            } else if (pq.top() < i) {
                pq.push(i);
                pq.pop();
            }
        }
        return pq.top();
    }
};

int main() {
    struct TestCase {
        vector<int> nums;
        int k;
    };

    vector<TestCase> testCases = {{{3, 2, 1, 5, 6, 4}, 2}};
    Solution sol;
    for (auto testCase : testCases) {
        cout << sol.findKthLargest(testCase.nums, testCase.k) << endl;
    }
    return 0;
}